mypy nox session fails with types-protobuf 3.19.8

See output [here](https://github.com/googleapis/gapic-generator-python/runs/5042352544?check_suite_focus=true):
```
Run nox -s mypy
nox > Running session mypy
nox > Creating virtual environment (virtualenv) using python3.9 in .nox/mypy
nox > python -m pip install mypy types-protobuf types-PyYAML types-dataclasses
nox > python -m pip install .
nox > mypy gapic
gapic/schema/api.py:130: error: "FileOptions" has no attribute "Extensions"
gapic/schema/api.py:881: error: "MethodOptions" has no attribute "Extensions"
Found 2 errors in 1 file (checked 35 source files)
nox > Command mypy gapic failed with exit code 1
nox > Session mypy failed.
```

PR #1179 adds a pin for `types-protobuf<=3.19.7` to allow checks to pass so they don't block development.

--- a/noxfile.py
+++ b/noxfile.py
@@ -380,7 +380,7 @@ def showcase_mypy(
     """Perform typecheck analysis on the generated Showcase library."""
 
     # Install pytest and gapic-generator-python
-    session.install("mypy", "types-pkg-resources", "types-protobuf<=3.19.7", "types-requests", "types-dataclasses")
+    session.install("mypy", "types-pkg-resources", "types-protobuf", "types-requests", "types-dataclasses")
 
     with showcase_library(session, templates=templates, other_opts=other_opts) as lib:
         session.chdir(lib)
@@ -443,6 +443,6 @@ def docs(session):
 def mypy(session):
     """Perform typecheck analysis."""
 
-    session.install("mypy", "types-protobuf", "types-PyYAML", "types-dataclasses")
+    session.install("mypy", "types-protobuf<=3.19.7", "types-PyYAML", "types-dataclasses")
     session.install(".")
     session.run("mypy", "gapic")

--- a/noxfile.py
+++ b/noxfile.py
@@ -380,7 +380,7 @@ def showcase_mypy(
     """Perform typecheck analysis on the generated Showcase library."""
 
     # Install pytest and gapic-generator-python
-    session.install("mypy", "types-pkg-resources", "types-protobuf", "types-requests", "types-dataclasses")
+    session.install("mypy", "types-pkg-resources", "types-protobuf<=3.19.7", "types-requests", "types-dataclasses")
 
     with showcase_library(session, templates=templates, other_opts=other_opts) as lib:
         session.chdir(lib)

--- a/requirements.txt
+++ b/requirements.txt
@@ -7,3 +7,4 @@ protobuf==3.19.4
 pypandoc==1.7.2
 PyYAML==6.0
 dataclasses==0.6  # TODO(busunkim) remove when 3.6 support is dropped
+setuptools<=60.5.0 # pin setuptools to workaround https://github.com/pypa/setuptools/issues/3072

